beacon_chain/utils/bls/src/aggregate_signature.rs

```diff
@@ -0,0 +1,72 @@
+use super::ssz::{decode_ssz_list, Decodable, DecodeError, Encodable, SszStream};
+use super::{AggregatePublicKey, Signature};
+use bls_aggregates::AggregateSignature as RawAggregateSignature;
+
+/// A BLS aggregate signature.
+///
+/// This struct is a wrapper upon a base type and provides helper functions (e.g., SSZ
+/// serialization).
+#[derive(Debug, PartialEq, Clone)]
+pub struct AggregateSignature(RawAggregateSignature);
+
+impl AggregateSignature {
+    /// Instantiate a new AggregateSignature.
+    pub fn new() -> Self {
+        AggregateSignature(RawAggregateSignature::new())
+    }
+
+    /// Add (aggregate) a signature to the `AggregateSignature`.
+    pub fn add(&mut self, signature: &Signature) {
+        self.0.add(signature.as_raw())
+    }
+
+    /// Verify the `AggregateSignature` against an `AggregatePublicKey`.
+    ///
+    /// Only returns `true` if the set of keys in the `AggregatePublicKey` match the set of keys
+    /// that signed the `AggregateSignature`.
+    pub fn verify(&self, msg: &[u8], avk: &AggregatePublicKey) -> bool {
+        self.0.verify(msg, avk)
+    }
+}
+
+impl Default for AggregateSignature {
+    /// A "default" signature is a signature across an empty message by a secret key of 48 zeros.
+    fn default() -> Self {
+        AggregateSignature::new()
+    }
+}
+
+impl Encodable for AggregateSignature {
+    fn ssz_append(&self, s: &mut SszStream) {
+        s.append_vec(&self.0.as_bytes());
+    }
+}
+
```

Comment:
in general i do like having the `ssz` logic for a type in the module where that type is defined. then we can do very minimal byte munging in higher level places like `Attestation`. this trend over time puts us in a place where we could even think about a custom serde backend for `ssz` (there is the beginning of one in `parity/shasper` but until the spec finalizes i suppose no one can publish an impl...) where we do simple "this is a list so just ssz each elem" or "this is a container so ssz each field" type logic and then defer to the leaves of this tree for special logic (like we have here)

Reply:
Agreed about a serde backend. Not a huge fan of the SSZ crate we have atm.

Reply:
Just to confirm, you're not suggesting a change in this comment, correct?